06-building-a-program-start-to-finish/writing-tests/Output/test/spec/test.js

```diff
@@ -0,0 +1,83 @@
+const chai = require('chai');
+const expect = chai.expect;
+ 
+ const InvertedIndex = require('../../src/js/inverted-index.js');
+ const books = require('../books.json');
+ 
+ const emptyBook = [];
+ const newIndex = new InvertedIndex();
+ const sampleData = ' Mary had a    little #$%^6 lamb, a little lamb Mary had';
+ newIndex.createIndex('books.json', books);
+ 
+ describe('Validate files', () => {
+   it('should check that uploaded file is valid JSON', () => {
+     expect(Array.isArray(books)).to.be.true;
+     expect(Array.isArray(emptyBook)).to.be.true;
+     expect(typeof books[0]).to.equal('object');
+   });
+ 
+   it('should not be empty', () => {
+     expect(books.length).to.not.equal(0);
+     expect(emptyBook.length).to.equal(0);
+   });
+   it("should have a valid 'title' and 'text'", () => {
+     expect(books[0].title).to.not.be.undefined;
+     expect(books[0].text).to.not.be.undefined;
+   });
+ });
+ 
+ describe('Normalized Text', () => {
+   it('should return an array containing only alphabets', () => {
+     expect(newIndex.normalizedText(sampleData)).to.not.include(' ');
+     expect(newIndex.normalizedText(sampleData)).to.not.include('#$%^6');
+   });
+   it('should return an array containing the correct number of words', () => {
+     expect(newIndex.normalizedText(sampleData).length).to.equal(10);
+   });
+ });
+ describe('Unique words', () => {
+   it('should not return any duplicate words', () => {
+     expect(newIndex.uniqueWords(sampleData).length).to.equal(5);
+   });
+ });
+ describe('Read Book Data', () => {
+   it('should check that JSON file is not empty', () => {
+     expect(newIndex.createIndex('emptyBooks.json', emptyBook))
+     .to.eql('JSON file is Empty');
+     expect(newIndex.createIndex('Books.json', books))
+     .to.not.eql('JSON file is Empty');
+   });
+ });
+ 
+ describe('Populate index', () => {
+   it('should create index once JSON file has been read', () => {
+     expect(newIndex.indices.length).to.not.equal(0);
+   });
+   it('should map words to the correct document location', () => {
+     expect(newIndex.indices['books.json'].fellowship).to.eql([1, 2]);
```

Comment:
Why are you calling the `indices` property directly here?

I believe your `getIndex` method is supposed to help with that.

Also, you're only testing for one word. I'd like to see you test for an exact complete result of the `createIndex`.